src/modules/esl-share/core/esl-share-list.ts

```diff
@@ -0,0 +1,96 @@
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, bind, boolAttr, memoize, prop} from '../../esl-utils/decorators';
+
+import {ESLShareConfig} from './esl-share-config';
+import {ESLShareButton} from './esl-share-button';
+
+import type {ShareButtonConfig} from './esl-share-button';
+import type {ShareConfig} from './esl-share-config';
+
+export class ESLShareList extends ESLBaseElement {
+  public static is = 'esl-share-list';
+
+  public static register(): void {
+    ESLShareButton.register();
+    super.register();
+  }
+
+  /** Event to dispatch on ready state of {@link ESLShareList} */
+  @prop('esl:share:ready') public SHARE_READY_EVENT: string;
+
+  @attr({readonly: true}) public list: string;
+  @attr({readonly: true}) public group: string;
+  @attr({dataAttr: true}) public url: string;
+  @attr({dataAttr: true}) public title: string;
+
+  @boolAttr({readonly: true}) public ready: boolean;
+
+  protected _ready: Promise<void>;
+
+  public get alias(): string {
+    return (this.constructor as typeof ESLBaseElement).is;
+  }
```

Comment:
Just an idea to discuss...  `eslTagName` - 2 in one the marker of the library-specific structure and the reference to the native meaning

Reply:
TBD @ala-n to investigate the posibility of typed this.constructor